cmd/proxy/mocks_test.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/miekg/dns"
+)
+
+var ErrUnimplementedMock error = errors.New("unimplemented")
+
+// MockQUICStream satisfies the quic.Stream interface
+type MockQUICStream struct {
+	ID quic.StreamID
+
+	// InBuf holds the packed msg bytes coming in from the client
+	InBuf []byte
+	// OutBuf holds the packed msg bytes coming in from the backend (and out to client)
+	OutBuf []byte
+
+	Recursion bool
+
+	msg *dns.Msg
+}
+
+func NewMockQUICStream(id int, name, qType string, recursion bool, dnssec bool) *MockQUICStream {
+	qt, ok := dns.StringToType[qType]
+	if !ok {
+		panic("invalid qtype: " + qType)
+	}
+
+	msg := &dns.Msg{}
+	msg.SetQuestion(dns.Fqdn(name), qt)
+	msg.RecursionDesired = recursion
+	msg.SetEdns0(4096, dnssec)
+
+	wire, err := msg.Pack()
+	if err != nil {
+		panic(fmt.Errorf("pack query: %w", err))
+	}
+
+	return &MockQUICStream{
+		ID:     quic.StreamID(id),
+		InBuf:  wire,
+		OutBuf: make([]byte, 4096),
+
+		msg: msg,
+	}
+}
+
+func (mqs *MockQUICStream) GetMsg() *dns.Msg {
+	return mqs.msg
+}
+
+// ************************************
+// quic.ReceiveStream interface methods
+// ************************************
+
+// StreamID returns the stream ID.
+func (mqs *MockQUICStream) StreamID() quic.StreamID {
+	return mqs.ID
+}
+
+// Read reads data from the stream.
+// Read can be made to time out and return a net.Error with Timeout() == true
```

Comment:
Please, do not copy documentation comments from the original interface description.